l10n_it_fatturapa_out_taxes/wizard/wizard_export_fatturapa.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Davide Corio
+# Copyright 2015-2016 Lorenzo Battistini - Agile Business Group
+# Copyright 2018 Gianmarco Conte, Marco Calcagni - Dinamiche Aziendali srl
+# Copyright 2018 Sergio Corato
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import logging
+
+from openerp import models, _
+from openerp.exceptions import Warning as UserError
+
+from openerp.addons.l10n_it_fatturapa.bindings.fatturapa_v_1_2 import (
+    ScontoMaggiorazioneType,
+    DettaglioLineeType,
+    CodiceArticoloType,
+    DatiBeniServiziType,
+    DatiRiepilogoType,
+)
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from unidecode import unidecode
+except ImportError as err:
+    _logger.debug(err)
+
+
+class WizardExportFatturapa(models.TransientModel):
+    _inherit = "wizard.export.fatturapa"
+
+    def setDettaglioLinee(self, invoice, body):
```

Comment:
Potresti chiamare super e modificare `DettaglioLinee` a posteriori, eventualmente

Reply:
essendo che ritorna True, faccio il super ma ripulisco tutto per proseguire

Reply:
Forse però il super andrebbe in eccezione per via di uno di questi 2 `if`
```
            if not taxes:
                raise UserError(
                    _("Invoice line %s does not have vat tax.") % line.name)
            if len(taxes) > 1:
                raise UserError(
                    _("Too many vat taxes for invoice line %s.") % line.name)
```
?

Ma qual è il tipico caso di imposta da escludere dai registri IVA?

Reply:
ritenute Enasarco

Reply:
Nelle fatture attive? Cioè quando l'utilizzatore di odoo è un agente che emette fattura con ritenuta?

Reply:
Si

Reply:
ho tolto entrambi i super(), visto che ci sono raise() su entrambe le funzioni